providers/base/bin/watchdog_config_test.py

```diff
@@ -54,44 +70,55 @@ def watchdog_service_check():
 
 
 def main():
```

Comment:
This is very convoluted because to understand what this is going to return you need to track `watchdog_config_ready` and at the end it is even negated. I think this has to become more functional. Something like
```python
if args.check_time:
    watchdog_config_ready = check_time(...)
elif args.check_service:
    watchdog_config_ready = check_service(...)
else:
    raise SystemExit("Either select check_time or check_service")
return not watchdog_config_ready
```